andela/metis-ah
# #159987421 role based access

#### What does this PR do?
Implements a role-based access control to routes
#### Description of Task to be completed?
- Create the `roles` model
- Add a `roleId` field to the user's table and establish a relationship with `roles`
- Create the following endpoints
  * `[GET] All roles`
  * `[POST] New role`
  * `[PUT] Edit role`
- Add middleware that checks role access to various routes
- Updates the documentation
#### How should this be manually tested?
1. Clone the repository.
2. Checkout the `ft-role-based-access-159987421` branch.
3. Run npm install on your local machine.
4. Run the following commands 
```export NODE_ENV=development
export DATABASE_URL=(Your-Local-DB-URL)
export TEST_DATABASE_URL=(Your-Local-Test-DB-URL)
npm run migrate
```
5. Start the server using `npm run start_dev`.
6. Sigin up and try to access `localhost:8000/api/v1/cases`.
7. You should get a `401` response with a `Only an admin can access this resource`
#### What are the relevant pivotal tracker stories?
#159987421
